flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BigQueryDefaultWriter.java

```diff
@@ -43,25 +42,33 @@
  * <p>Depending on the checkpointing mode, this writer offers the following consistency guarantees:
  * <li>{@link CheckpointingMode#EXACTLY_ONCE}: at-least-once write consistency.
  * <li>{@link CheckpointingMode#AT_LEAST_ONCE}: at-least-once write consistency.
- * <li>{Checkpointing disabled}: no write consistency.
+ * <li>Checkpointing disabled: no write consistency.
  *
  * @param <IN> Type of records to be written to BigQuery.
  */
 public class BigQueryDefaultWriter<IN> extends BaseWriter<IN> {
 
     public BigQueryDefaultWriter(
             int subtaskId,
+            String tablePath,
             BigQueryConnectOptions connectOptions,
             BigQuerySchemaProvider schemaProvider,
-            BigQueryProtoSerializer serializer,
-            String tablePath) {
-        super(subtaskId, connectOptions, schemaProvider, serializer);
+            BigQueryProtoSerializer serializer) {
+        super(subtaskId, tablePath, connectOptions, schemaProvider, serializer);
         streamName = String.format("%s/streams/_default", tablePath);
+        totalRecordsSeen = 0L;
+        totalRecordsWritten = 0L;
     }
 
-    /** Accept record for writing to BigQuery table. */
+    /**
+     * Accept record for writing to BigQuery table.
+     *
+     * @param element Record to write
+     * @param context {@link Context} for input record
+     */
     @Override
     public void write(IN element, Context context) {
+        totalRecordsSeen++;
```

Comment:
this might be out of scope for this PR but it might be worth trcking the element sizes here as well

Reply:
ie, totalRecordsSeenSize

Reply:
@prashastia this should go into our document tracking Flink metrics